### ISSUE TITLE
Are there docs/examples for the logging functionality?

### ISSUE THREAD

**Issue by Falmarri** [open-telemetry/opentelemetry-cpp#2302]
The last release marks logging as public. But I don't see any docs about this. Is this intended to be a general purpose logging framework? I'm currently using spdlog, but I have structure hacked into it. If this project also is intending to support native structured logging, I'm very interested. But I'm hesitant given there seems to be 0 documentation.

**Comment by marcalff** [comment]
This is missing, and part of the plan ...
* #1695

**Comment by ThomsonTan** [comment]
@Falmarri , our API docs are still under progress. Our API is intended for general purpose logging, but currently we only support OpenTelemetry backend (SDK&exporter). Wondering which logging scenario you are interested in.

**Comment by Falmarri** [comment]
@ThomsonTan Basically structured logging to files, with arbitrary key/value pairs. Specifically we log in [ECS format](https://www.elastic.co/guide/en/ecs/8.11/ecs-field-reference.html). But also the ability to use arbitrary key values. I currently hack this using `spdlog` and some custom json. Ideally it would work somewhat similar to python logs (with the knowledge that python is much more flexible here), such that some form/version of

`logger.info("Some message", key="value", **{"event.duration": 1000})`

would result in

`{"@timestamp": ..., "log": {"level": "info", "logger": "whatever"}, "message": {"Some message"}, "event": {"duration": 1000}, "key": "value"}`

Or some similar syntax.

Basically, just files and/or stdout. We have a lot of logging framework built to eventually get ingested by our ELK stack.
